packages/cli/src/commands/contract/call.ts

```diff
@@ -1,49 +1,87 @@
 import { Command, Flags } from "@oclif/core";
-import { execSync } from "node:child_process";
 import path = require("node:path");
-import { getSwankyConfig, resolveNetworkUrl } from "@astar-network/swanky-core";
+import { cryptoWaitReady } from "@polkadot/util-crypto";
+import { ContractPromise } from "@polkadot/api-contract";
+import {
+  AccountData,
+  ChainAccount,
+  ChainApi,
+  decrypt,
+  Encrypted,
+  getSwankyConfig,
+  resolveNetworkUrl,
+  Spinner,
+  AbiType as Abi,
+} from "@astar-network/swanky-core";
+import inquirer from "inquirer";
+import chalk = require("chalk");
+import { readJSON } from "fs-extra";
 
 export class CallContract extends Command {
   static description = "Call a method on a smart contract";
 
+  static args = [
+    {
+      name: "contractName",
+      required: true,
+      description: "Name of the contract to call",
+    },
+    {
+      name: "messageType",
+      required: true,
+      description: "Is the call a query or tx type",
+      options: ["q", "query", "tx"],
+    },
+  ];
+
   static flags = {
-    args: Flags.string({
+    params: Flags.string({
       required: false,
-      char: "a",
+      description: "Arguments supplied to the message",
+      multiple: true,
+      default: [],
+      char: "p",
     }),
-    contractName: Flags.string({
+    account: Flags.string({
       required: true,
+      char: "a",
+      description: "Account to sign the transaction with",
     }),
     message: Flags.string({
       required: true,
       char: "m",
+      description: "Message name to call",
     }),
     dry: Flags.boolean({
       char: "d",
+      description: "Do a dry run, without signing the transaction (only for tx type of message)",
     }),
     gas: Flags.string({
       char: "g",
+      description: "Manually specify gas limit",
     }),
     network: Flags.string({
       char: "n",
       description: "Network name to connect to",
     }),
-    deploymentTimestamp: Flags.integer({
```

Comment:
We lost the way to choose which contracts to call in case several deployments we have.
So, we need to recover some alternatives.

I suggest `contractAddress` or `contractAccountId`, because it's universal concept in blockchain space.
And users don't need to convert something (e.g. timestamp) to a contract address in their head when trying to look up in explorers or some external tools to check.

Reply:
Added `address` flag in the update